modules/ROOT/pages/indexes/semantic-indexes/vector-indexes.adoc

```diff
@@ -1,503 +1,302 @@
-
-
-.Syntax for creating vector indexes
-[options="header", width="100%", cols="5a, 3"]
-|===
-| Command | Description
-
-| [source, syntax, role=noplay, indent=0]
-----
-CREATE VECTOR INDEX [index_name] [IF NOT EXISTS]
-FOR (n:LabelName)
-ON (n.propertyName)
-OPTIONS "{" option: value[, ...] "}"
+.Create vector index for `Movie` nodes on the `embedding` property label:new[Introduced in 5.15]
+[source, cypher]
 ----
-| Create a vector index on nodes.
-The options map is mandatory because setting the vector dimensions and similarity function is mandatory when creating a vector index.
-
-| [source, syntax, role=noplay, indent=0]
-----
-CREATE VECTOR INDEX [index_name] [IF NOT EXISTS]
-FOR ()-”[“r:TYPE_NAME”]”-()
-ON (r.propertyName)
-OPTIONS "{" option: value[, ...] "}"
+CREATE VECTOR INDEX moviePlots IF NOT EXISTS // <1>
+FOR (m:Movie)
+ON m.embedding
+OPTIONS {indexConfig: { 
+ `vector.dimensions`: 1536,
+ `vector.similarity_function`: 'cosine'
+}} // <2>
 ----
-| Create a vector index on relationships.
-The options map is mandatory because setting the vector dimensions and similarity function is mandatory when creating a vector index.  label:new[Introduced in 5.18]
 
-|===
-
-It is considered best practice to give the index a name when it is created.
-This name is needed for both dropping and querying the index.
-If the index is not explicitly named, it will get an auto-generated name.
-As of Neo4j 5.16, the index name can also be given as a parameter, `CREATE VECTOR INDEX $name FOR ...`.
-
-[IMPORTANT]
-====
-The index name must be unique among all indexes and constraints.
-====
-
-The `CREATE VECTOR INDEX` command take an `OPTIONS` clause. This has two parts, the `indexProvider` and `indexConfig`.
-As of Neo4j 5.18, there are two available index providers, `vector-2.0` (default) and `vector-1.0`.
-The `indexConfig` is a `MAP` from `STRING` values to `STRING` and `INTEGER` values, and is used to set the index-specific configuration settings (the vector dimensions and similarity function).
-
-The command is optionally idempotent. This means that its default behavior is to throw an error if an attempt is made to create the same index twice.
+<1> The `CREATE VECTOR INDEX` command is optionally idempotent.
+This means that its default behavior is to throw an error if an attempt is made to create the same index twice.
 With `IF NOT EXISTS`, no error is thrown and nothing happens should an index with the same name, schema or both already exist.
 It may still throw an error should a constraint with the same name exist.
 As of Neo4j 5.17, an informational notification is instead returned showing the existing index which blocks the creation.
+<2> The `OPTIONS` map is mandatory since a vector index cannot be created without setting the vector dimensions and similarity function.
+In this example, the vector dimension is set to `1536` and the vector similarity function is `cosine`, which generally perfoms best for text embeddings.
```

Comment:
This is what we advice in the GraphAcademy, and I have read the same heuristic elsewhere (e.g. https://www.geeksforgeeks.org/when-to-use-cosine-similarity-over-euclidean-similarity/ - "In summary, use Cosine Similarity when dealing with high-dimensional data, text analysis, or situations where the magnitude of vectors is not crucial", or https://www.baeldung.com/cs/euclidean-distance-vs-cosine-similarity -
" retrieval of the most similar texts to a given document, generally function better with cosine similarity"). 

I am aware that these are not academic references, and I can remove it if you think it is not accurate. My thinking in including it was that it might be a useful guideline for non-experts reading along, who can then at a later stage familiarise themselves with your detailed explanation of the differences between the two, which is still in the document.

Reply:
That's fair. It is often a safe bet to get something, somewhat useful.

The reason we use cosine with open ai, not only because its what they suggest, but because it's a unit vector (thus magnitude doesn't matter) and that's cosine's bread and butter.

I wouldn't say it's a wrong statement, the main issue I have is to use of "perform" without context, as that means many different things to different people.

Perhaps, something more like: "which is generally the preferred similarity function for text embeddings" to the need for the context, whilst still suggesting it's a good default option to go for.